pkg/chunk/schema.go

```diff
@@ -204,6 +208,67 @@ func (s schema) GetChunksForSeries(from, through model.Time, userID string, seri
 	return result, nil
 }
 
+// returns IndexEntrys for deletion. It makes sure that we don't include series entries which are in use by verifying using hasChunksForIntervalFunc
+func (s schema) GetSeriesDeleteEntries(from, through model.Time, userID string, metric labels.Labels, hasChunksForIntervalFunc hasChunksForIntervalFunc) ([]IndexEntry, error) {
```

Comment:
`SeriesID` is stored once per bucket so for partial deletion, we don't want to delete those entries otherwise rest of the chunks won't be accessible. Will add a comment. Thanks!